### PR TITLE
copy manager addrs with DOMAIN_NAME_MAX limit

### PR DESCRIPTION
**Pull request by btovar**
## Proposed changes

Please describe your changes (e.g., what problems they attempt to solve, what results are expected, etc.) Additional motivation and context are welcome.
Please also mention relevant issues and pull requests as appropriate.

## Post-change actions

Put an 'x' in the boxes that describe post-change actions that you have done.
The more 'x' ticked, the faster your changes are accepted by maintainers.

- [x] `make test`       Run local tests prior to pushing.
- [ ] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [x] `make lint`       Run lint on source code prior to pushing.
- [ ] Manual Update     Did you update the manual to reflect your changes, if appropriate? This action should be done after your changes are approved but not merged.
- [x] Type Labels       Select github labels for the type of this change: bug, enhancement, etc.
- [x] Product Labels    Select github labels for the product affected: TaskVine, Makeflow, etc.
- [x] PR RTM            Mark your PR as ready to merge.

## Additional comments
This section is dedicated to changes that are ambitious or complex and require substantial discussions. Feel free to start the ball rolling.

### CODE REVIEW

**Comment by dthain** [comment]
Something still doesn't add up there.
Maybe it's just the names of variables.
Anything `addr` should refer to an Internet address, IPV4 or IPV6.
Anything `host` or `name` should refer to a domain name.

```
struct manager_address {
        char host[DOMAIN_NAME_MAX];
        int port;
        char addr[DOMAIN_NAME_MAX];
};
```

The `host` field is correct but the `addr` field should be `LINK_ADDR_MAX`.

And then this line is confusing because it appears to be copying a interface address into a domain name:
```
strncpy(m->host, ifa_addr, DOMAIN_NAME_MAX - 1);
```

But `ifa_addr` comes from `jx_lookup_string(ifa,host)`

I think we are getting confused b/c variables names do not clearly reflect whether something is a domain name or an address.

**Comment by btovar** [comment]
Right, we are copying ifa_addr to host in order to force a particular network interface. 

I agree that it should be `addr[LINK_ADDR_MAX]`, but that only wastes some bits.

**Comment by dthain** [comment]
Sorry, I wasn't very clear, let me try again:

IMHO, the fundamental problem here isn't the length of the `addr` array. 

The problem is that the code doesn't make it clear which variables are addresses, and which ones are hostnames. The former can be passed to `link_connect` etc, and the latter must be passed to `domain_name_lookup`, and they cannot be trivially exchanged. I think if you take a few moments to clean up the names of those variables, then the correctness (or not) of the code will become obvious.

This is made worse by the fact that `interfaces_to_list` can take either a hostname or an address as its first argument, because it deals with the various connect options advertised by the manager:
https://github.com/cooperative-computing-lab/cctools/blob/ce4affa74399bb8301786dff4d263964b21fd12e/taskvine/src/worker/vine_worker.c#L1869

And so perhaps that argument should be `host_or_addr` in order to be abundantly clear.
